### PR TITLE
Rename example template to not use pod-style path

### LINKED ISSUE #614: Components > Composing Components Section inconsistently has pod based example
In the [Data Down](http://guides.emberjs.com/v2.0.0/components/composing-components/#toc_data-down) paragraph/subsection it introduces a template with a pod-based filepath for (I believe) the first time if sequentially reading the guides.

`app/users/template.hbs`

Wouldn't that be more consistent to be `app/templates/users.hbs`?

### PR DESCRIPTION
**Pull request by gjastrab**
Fixes #614

### CODE REVIEW

File: source/components/composing-components.md [hunk 1]
```diff
@@ -110,7 +110,7 @@ with the `as` operator. Let's take the following template as an example:
 This `{{user-list}}` component will yield as many times as there are users.
 We can consume it in the following way:
 
-```app/users/template.hbs
+```app/templates/index.hbs
```
Comment (locks) [inline source/components/composing-components.md]: This should be `app/templates/application.hbs` like the previous examples.

**Comment by locks** [comment]
There are still two `app/users/template.hbs` paths in the file, can you address those as well? Appreciated.

**Comment by gjastrab** [comment]
repushed with those changes

**Comment by michaelrkn** [comment]
While you're at it, can you [rename all pod-style paths in this file](https://github.com/emberjs/guides/issues/614#issuecomment-132516800), eg `app/components/user-list/template.hbs` -> `app/templates/components/user-list.hbs`? Thanks for your help, @gjastrab!

**Comment by locks** [comment]
I read the whole file and still missed those, oops!

**Comment by gjastrab** [comment]
Ok repushed again.  In addition to `app/components/user-list/template.hbs` fix there was also a case of `app/components/user-profile/component.js` which I changed to `app/components/user-profile.js`

**Comment by locks** [comment]
LGTM, thanks for your contribution!

**Comment by gjastrab** [comment]
My pleasure to help!

**Pull request merged**